app/helpers/ExerciseConfig/Pipeline/Box/Boxes/base/JavaUtils.php

```diff
@@ -9,20 +9,32 @@ class JavaUtils
     const CURRENT_DIR = ".";
     const PATH_DELIM = ":";
 
-    public static function constructClasspath(?Variable $jarFiles)
+    public static function constructClasspath(?Variable $jarFiles, ?string $compiledClassesDirectory = null, ?Variable $classpath = null)
     {
+        $result = [];
+
+        // jar filed specified by exercise author
         if ($jarFiles && !$jarFiles->isEmpty()) {
-            $classpath = JavaUtils::CURRENT_DIR;
             foreach ($jarFiles->getValueAsArray() as $jar) {
-                $classpath .= JavaUtils::PATH_DELIM . $jar;
+                $result[] = $jar;
             }
+        }
+
+        if (!empty($compiledClassesDirectory)) {
+            $result[] = $compiledClassesDirectory;
+        }
+
+        // might be used for worker local jar files (groovy stdlib or kotlin stdlib)
+        if ($classpath && !$classpath->isEmpty()) {
+            foreach ($classpath->getValueAsArray() as $cp) {
+                $result[] = $cp;
+            }
+        }
 
-            return [
-                "-classpath",
-                $classpath
-            ];
+        if (empty($result)) {
+            return [];
         }
 
-        return [];
+        return ["-classpath", JavaUtils::CURRENT_DIR . self::PATH_DELIM . join(self::PATH_DELIM, $result)];
```

Comment:
IMHO, it would be more elegant to place JavaUtils::CURRENT_DIR as the initial value of $result array and then use only join (no string concatenation) here.

Reply:
Well, I was being lazy and did not know at the moment how to prepend element to array...